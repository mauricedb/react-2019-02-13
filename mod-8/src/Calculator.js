class Calculator {
  add(x, y) {
    return x + y;
  }

  substract(x, y) {
    return x - y;
  }
}
