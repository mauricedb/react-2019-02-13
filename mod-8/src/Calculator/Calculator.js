import { log } from './logger';

class Calculator {
  add(x, y) {
    log('Ik zit in add');
    return x + y;
  }

  substract(x, y) {
    return x - y;
  }
}

export default Calculator;
