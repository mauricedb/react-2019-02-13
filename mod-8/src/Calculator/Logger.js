export function log(msg) {
  throw new Error('Dit mag niet');
}
