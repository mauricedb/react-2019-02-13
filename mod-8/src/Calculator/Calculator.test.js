import Calculator from './Calculator';

jest.mock('./logger');

test('een rekenmachine kan 1 + 1 optellen', () => {
  const calc = new Calculator();

  const sum = calc.add(1, 1);

  expect(sum).toBe(2);
});

test('een rekenmachine kan 0.1 + 0.2 optellen', () => {
  const calc = new Calculator();

  const sum = calc.add(0.1, 0.2);

  expect(sum).toBeCloseTo(0.3, 15);
});

test('snapshot', () => {
  const calc = new Calculator();

  const sum = calc.add(0.1, 0.2);

  expect(sum).toMatchSnapshot();

  expect([{ theSum: sum }]).toMatchSnapshot();
});

test('een rekenmachine kan true + false optellen', () => {
  const calc = new Calculator();

  const sum = calc.add(true, false);

  expect(sum).toBe(1);
});

test('een rekenmachine kan [] + [] optellen', () => {
  const calc = new Calculator();

  const sum = calc.add([], []);

  expect(sum).toBe('');
});
