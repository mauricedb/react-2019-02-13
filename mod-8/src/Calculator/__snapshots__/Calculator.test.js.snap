// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `0.30000000000000004`;

exports[`snapshot 2`] = `
Array [
  Object {
    "theSum": 0.30000000000000004,
  },
]
`;
